src/components/GroupSystems/GroupImmutableSystems.js

```diff
@@ -131,7 +131,6 @@ const GroupImmutableSystems = ({ groupName, groupId, ...props }) => {
       enhancedConfig,
       showTags
     );
-
     const mapDeviceIds = mapDefaultData(defaultData.results);
     const updateInfo = await getUpdateInfo(groupId);
     setDeviceData(updateInfo?.update_devices_uuids);
```

Comment:
nit-pick:
```suggestion
    setDeviceData(updateInfo?.update_devices_uuids || []);
```

since the server responds with `null` if there are no systems pending updates, I would recommend setting deviceData to [] so that it's more safe to work with this state variable across the component. For example, in that case, you'd rewrite the "Update" button condition (if it's disabled or not) to just `!deviceData.includes(row.id)`

Reply:
Changed for your suggestion and rebased the branch to help on merge.

Thanks for your review